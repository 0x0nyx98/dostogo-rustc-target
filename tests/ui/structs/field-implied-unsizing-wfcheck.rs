struct Foo {
    nested: &'static Bar<dyn std::fmt::Debug>,
    //~^ ERROR the size for values of type `(dyn Debug + 'static)` cannot be known at compilation time
}

struct Bar<T>(T);

fn main() {
    let x = Foo { nested: &Bar(4) };
    //~^ ERROR the size for values of type `(dyn Debug + 'static)` cannot be known at compilation time
}
