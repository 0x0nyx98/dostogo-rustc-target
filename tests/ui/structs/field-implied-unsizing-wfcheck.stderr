error[E0277]: the size for values of type `(dyn Debug + 'static)` cannot be known at compilation time
  --> $DIR/field-implied-unsizing-wfcheck.rs:2:13
   |
LL |     nested: &'static Bar<dyn std::fmt::Debug>,
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `(dyn Debug + 'static)`
note: required by an implicit `Sized` bound in `Bar`
  --> $DIR/field-implied-unsizing-wfcheck.rs:6:12
   |
LL | struct Bar<T>(T);
   |            ^ required by the implicit `Sized` requirement on this type parameter in `Bar`
help: you could relax the implicit `Sized` bound on `T` if it were used through indirection like `&T` or `Box<T>`
  --> $DIR/field-implied-unsizing-wfcheck.rs:6:12
   |
LL | struct Bar<T>(T);
   |            ^  - ...if indirection were used here: `Box<T>`
   |            |
   |            this could be changed to `T: ?Sized`...

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
